File: holoken-core/src/main/java/com/holokenmod/creation/RandomPossibleDigitsShuffler.java
package com.holokenmod.creation;

import com.holokenmod.RandomSingleton;

import java.util.Collections;
import java.util.List;

public class RandomPossibleDigitsShuffler implements PossibleDigitsShuffler {
	@Override
	public void shufflePossibleDigits(List<Integer> possibleDigits) {
		if (!possibleDigits.isEmpty()) {
			Collections.shuffle(possibleDigits, RandomSingleton.getInstance().getRandom());
		}
	}
}

File: holoken-core/src/main/kotlin/com/holokenmod/creation/RandomPossibleDigitsShuffler.kt
package com.holokenmod.creation

import com.holokenmod.RandomSingleton

class RandomPossibleDigitsShuffler : PossibleDigitsShuffler {
    override fun shufflePossibleDigits(possibleDigits: Set<Int>): List<Int> {
        return possibleDigits.shuffled(RandomSingleton.instance.getRandom())
    }
}
